app/src/main/java/ly/loud/loudly/util/BasePresenter.java

```diff
@@ -0,0 +1,111 @@
+package ly.loud.loudly.util;
+
+import android.support.annotation.AnyThread;
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.UiThread;
+import android.support.v4.app.Fragment;
+
+import ly.loud.loudly.application.Loudly;
+import rx.Subscription;
+import rx.functions.Action1;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Base class for all presenters.
+ * @param <V> - view to be bounded to
+ */
+public class BasePresenter<V> {
+
+    /**
+     * Base application Context.
+     */
+    @NonNull
+    private Loudly loudlyApplication;
+
+    /**
+     * Bound view or null if no view is currently bounded to this presenter.
+     */
+    @Nullable
+    private volatile V view;
```

Comment:
presenter works with defferent threads so we need atomic access to it in method such as getView()